### PR TITLE
fix: asset-swapper yield to the event loop every mix path

### PR DESCRIPTION
**Pull request by dekz**
## Description

Yield to the event loop during CPU intensive tasks. Adds a little overheard but allows other tasks to be worked on (i.e move something to waiting to i/o waiting)

### CODE REVIEW

**Review by opaolini — APPROVED** [review]

File: packages/asset-swapper/src/utils/market_operation_utils/path_optimizer.ts [hunk 1]
```diff
@@ -9,24 +9,29 @@ import { Fill } from './types';
...
-): Fill[] | undefined {
+): Promise<Fill[] | undefined> {
     // Sort paths in descending order by adjusted output amount.
     const sortedPaths = paths
         .slice(0)
         .sort((a, b) => getPathAdjustedSize(b, targetInput)[1].comparedTo(getPathAdjustedSize(a, targetInput)[1]));
     let optimalPath = sortedPaths[0] || [];
     for (const [i, path] of sortedPaths.slice(1).entries()) {
         optimalPath = mixPaths(side, optimalPath, path, targetInput, runLimit * RUN_LIMIT_DECAY_FACTOR ** i);
+        await setImmediateAsync();
```
Comment (dorothy-zbornak) [inline packages/asset-swapper/src/utils/market_operation_utils/path_optimizer.ts:34]: :trollface: 
```suggestion
        await null;
```

**Review by dorothy-zbornak — APPROVED** [review]

File: packages/asset-swapper/CHANGELOG.json [hunk 2]
```diff
@@ -1,4 +1,13 @@
 [
+    {
+        "version": "4.6.1",
+        "changes": [
+            {
+                "note": "Allow an empty override for sampler overrides",
```
Comment (dorothy-zbornak) [inline packages/asset-swapper/CHANGELOG.json:6]: Should prob also mention the async optimizer.

File: packages/asset-swapper/CHANGELOG.json [hunk 3]
```diff
@@ -1,4 +1,13 @@
...
+                "pr": 2635
```
Comment (dorothy-zbornak) [inline packages/asset-swapper/CHANGELOG.json]:
```suggestion
                "pr": 2637
```

**Review by moodlezoup — APPROVED** [review]
lgtm, thanks for fixing!

**Pull request merged**